tests/unit/src/Internal/Payment/State/InitialStateTest.php

```diff
@@ -254,4 +253,20 @@ public function test_populate_context_from_order() {
 
 		PHPUnit_Utils::call_method( $this->sut, 'populate_context_from_order', [] );
 	}
+
+	private function mock_initial_state() {
```

Comment:
Re-defining the service under tests like this:

1. Makes it a bit more hidden, and not so obvious what's being done, as it's in the end of the test.
2. Causes a bunch of errors when editing:

<img width="489" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5311119/03b347d0-6584-4c3a-985f-fb3e982bb98b">

As there are a multiple methods that use the pure class as `sut`, and multiple others that need the mocked one, let's just define both of them in the `setUp` method, something like this:

```php
/**
 * Service under test.
 *
 * @var InitialState
 */
private $sut;

/**
 * Service under test with mocked protected methods.
 *
 * @var InitialState|MockObject
 */
private $mocked_sut;

/**
 * Set up the test.
 */
protected function setUp(): void {
	parent::setUp();

	// set up mocks...

	$this->sut = new InitialState(
		$this->mock_state_factory,
		$this->mock_order_service,
		$this->mock_customer_service,
		$this->mock_level3_service,
		$this->mock_payment_request_service
	);
	$this->sut->set_context( $this->mock_context );

	$this->mocked_sut = $this->getMockBuilder( InitialState::class )
		->onlyMethods( [ 'populate_context_from_request', 'populate_context_from_order' ] )
		->setConstructorArgs(
			[
				$this->mock_state_factory,
				$this->mock_order_service,
				$this->mock_customer_service,
				$this->mock_level3_service,
				$this->mock_payment_request_service,
			]
		)
		->getMock();
	$this->mocked_sut->set_context( $this->mock_context );
}
```

Then tests can decide which one to use.